packages/core/src/actions.ts

```diff
@@ -808,28 +803,34 @@ export function resolveActions<TContext, TEvent extends EventObject>(
             }
           };
         }
-        return resolvedActionObject;
+        return [resolvedActionObject];
+      }
     }
   }
 
-  function processBlock(block: ActionObject<TContext, TEvent>[]) {
-    let resolvedActions: Array<ActionObject<TContext, TEvent>> = [];
-
-    for (const action of block) {
-      const resolved = handleAction(action);
-      if (resolved) {
-        resolvedActions = resolvedActions.concat(resolved);
-      }
-    }
+  function resolveActionBlocks(
+    _actionBlocks: ActionObject<TContext, TEvent>[]
```

Comment:
renamed because theres a variable with an equal name in the outer function. The overshadowing was noted in ESLint.
I probably would have preferred to move this method to the topmost level. But seems to not be the pattern elsewhere, so i decided to just rename the parameter.